pkg/sql/colexec/BUILD.bazel

```diff
@@ -200,57 +200,313 @@ go_test(
-    "hashtable_full_default.eg.go",
-    "hashtable_full_deleting.eg.go",
-    "hash_aggregator.eg.go",
-    "hash_utils.eg.go",
-    "is_null_ops.eg.go",
-    "like_ops.eg.go",
-    "mergejoinbase.eg.go",
-    "mergejoiner_exceptall.eg.go",
-    "mergejoiner_fullouter.eg.go",
-    "mergejoiner_inner.eg.go",
-    "mergejoiner_intersectall.eg.go",
-    "mergejoiner_leftanti.eg.go",
-    "mergejoiner_leftouter.eg.go",
-    "mergejoiner_leftsemi.eg.go",
-    "mergejoiner_rightanti.eg.go",
-    "mergejoiner_rightouter.eg.go",
-    "mergejoiner_rightsemi.eg.go",
-    "ordered_synchronizer.eg.go",
-    "proj_const_left_ops.eg.go",
-    "proj_const_right_ops.eg.go",
-    "proj_non_const_ops.eg.go",
-    "quicksort.eg.go",
-    "rank.eg.go",
-    "relative_rank.eg.go",
-    "row_number.eg.go",
-    "rowstovec.eg.go",
-    "selection_ops.eg.go",
-    "select_in.eg.go",
-    "sort.eg.go",
-    "substring.eg.go",
-    "values_differ.eg.go",
-    "vec_comparators.eg.go",
-    "window_peer_grouper.eg.go",
-]
-
-# Define a generator for all execgen code. Look towards COLEXEC.bzl for the
-# implementation.
-generate(
+filegroup(
     name = "gen-exec",
-    execgen = "//pkg/sql/colexec/execgen/cmd/execgen",
-    goimports = "//vendor/github.com/cockroachdb/gostdlib/x/tools/cmd/goimports",
-    targets = targets,
-    templates = glob(["*_tmpl.go"]),
+    srcs = [
+        ":gen-and-or-projection",
+        ":gen-cast",
+        ":gen-const",
+        ":gen-default-cmp-expr",
+        ":gen-default-cmp-proj-ops",
+        ":gen-default-cmp-sel-ops",
+        ":gen-distinct",
+        ":gen-hash-aggregator",
+        ":gen-hash-utils",
+        ":gen-hashjoiner",
+        ":gen-hashtable-distinct",
+        ":gen-hashtable-full-default",
+        ":gen-hashtable-full-deleting",
+        ":gen-is-null-ops",
+        ":gen-like-ops",
+        ":gen-mergejoinbase",
+        ":gen-mergejoiner-exceptall",
+        ":gen-mergejoiner-fullouter",
+        ":gen-mergejoiner-inner",
+        ":gen-mergejoiner-intersectall",
+        ":gen-mergejoiner-leftanti",
+        ":gen-mergejoiner-leftouter",
+        ":gen-mergejoiner-leftsemi",
+        ":gen-mergejoiner-rightanti",
+        ":gen-mergejoiner-rightouter",
+        ":gen-mergejoiner-rightsemi",
+        ":gen-ordered-synchronizer",
+        ":gen-proj-const-left-ops",
+        ":gen-proj-const-right-ops",
+        ":gen-proj-non-const-ops",
+        ":gen-quicksort",
+        ":gen-rank",
+        ":gen-relative-rank",
+        ":gen-row-number",
+        ":gen-rowstovec",
+        ":gen-select-in",
+        ":gen-selection-ops",
+        ":gen-sort",
+        ":gen-substring",
+        ":gen-values-differ",
+        ":gen-vec-comparators",
+        ":gen-window-peer-grouper",
+    ],
+)
+
+alias(
+    name = "execgen",
+    actual = "//pkg/sql/colexec/execgen/cmd/execgen",
+)
+
+alias(
+    name = "goimports",
+    actual = "//vendor/github.com/cockroachdb/gostdlib/x/tools/cmd/goimports",
+)
+
+generate(
```

Comment:
maybe for readability:
```
output = [
   ('and_or_projection', 'and_or_projection_tmpl.go'),
   ('cast', 'cast_tmpl.go')
]

srcs = [':gen-{}'.format(out) for out, _ in output]
for out, tmpl in output:
   generate(
      name='gen-' + out,
      output='{}.eg.go'.format(out),
      template=template,
   )
```

to make this easier to maintain?

Reply:
Done.